template-processors/base/bin/resourceManager.sh

```diff
@@ -80,6 +80,13 @@ function deleteByOldLabels() {
 function createUpdateResources() {
     local owner="$1"
     local timestamp="$(date +%s)"
+    # Check if directory contains only hidden files like .gitkeep, or .gitignore.
+    # This would mean that user purposefully wanted to track an empty directory in git.
+    # https://git.wiki.kernel.org/index.php/Git_FAQ#Can_I_add_empty_directories.3F
+    if [[ -z $(ls "${MANIFEST_DIR}") ]]; then
```

Comment:
I know the original issue made mention of an empty directory, but I'm wondering if maybe we want to check for a directory without any json/yaml/yml files:
```
$ ls randomfiles/
README.md  sa.yaml
$ kubectl apply -R -f randomfiles/
serviceaccount/my-service-account created
$ rm randomfiles/sa.yaml
$ ls randomfiles/
README.md
$ kubectl apply -R -f randomfiles/
error: error reading [randomfiles/]: recognized file extensions are [.json .yaml .yml]
$ echo $?
1
```

Reply:
While doing this PR I was also thinking about it at some point. In my opinion when something goes wrong with `kubectl apply -R -f`, the error message is descriptive enough for the user to know what might've gone wrong. E.g. in the example you provided the error message says that there were no json, yaml, or yml files and it's rather clear, I think. We could catch an error and wrap it in our own message but in the end the original error message from `kubectl` tells us the most about the problem:
```
$ ls -a broken/
.       ..      sth.yml
$ cat broken/sth.yml 
---
fake fake fake

$ kubectl apply -R -f broken/
error: error validating "broken/sth.yml": error validating data: invalid object to validate; if you choose to ignore these errors, turn validation off with --validate=false
```

Reply:
I agree, the `kubectl` command does a good job explaining what the error is. So I guess my question would be, rather than just checking if there is nothing in the directory:
`if [[ -z $(ls "${MANIFEST_DIR}") ]]; then`

What if we only applied if yaml/yml/json files exist in the directory:
`if [[ -z $(find ${MANIFEST_DIR} -name "*.yml" -o -name "*.json" -o -name "*.yaml") ]]; then`

If we think that's overkill, I'm ok with how it is. It would just help if a user stubbed in a README or some other file in the directory explaining what it's going to be used for before they added any manifests.

Reply:
I've added a check for yml, yaml, and json files. I created another if block because when directory is empty (has a .gitkeep or other hidden file) we don't want to raise an error (user wanted an empty directory explicitly), and when directory isn't empty but contains no yaml, yml, or json files it's an error.